### PR TITLE
Docusaurus v3 upgrade

### PR DESCRIPTION
**Pull request by kcpevey**
This PR upgrades the docs to Docusaurus V3. There are some [breaking changes](https://docusaurus.io/docs/migration/v3) and this PR includes all the required changes to make the docs build again after the upgrade. 

Some things to note:
* the `icon` admonition we had in the docusaurus config is no longer allowed. I had a hard time understanding the admonitions, but it seems like the `icon` admonition was being customized in `/src/theme/components/admonitions.scss` and the examples for V3 are all using js. Those are outside of my skillset so if we want to keep that, I'll need some help with that bit.
* I'm not very familiar with yarn, but I followed the directions to upgrade all the packages and update the `package.json` and `yarn.lock`. I _think_ I did that right?
* A lot of the required updates were around links. While I was fixing the link syntax, if the link was not accessible (i.e. was just underneath the word "here"), I reworded the text so the link text would be more descriptive. That is only reason for the wording changes in this PR. 

## Reference Issues or PRs

Resolves https://github.com/nebari-dev/nebari-docs/issues/395

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [x] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [ ] Did you add new tests?

#### Text styling

- [ ] The content is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [ ] If there are headers, they use the proper header tags (with only one level-one header: `H1` or `#` in markdown).
- [x] All links describe where they link to (for example, check the [Nebari website](https://nebari.dev/)).
- [ ] This content adheres to the Nebari style guides.

#### Non-text content

- [ ] All content is represented as text (for example, images need alt text, and videos need captions or descriptive transcripts).
- [ ] If there are emojis, there are not more than three in a row.
- [ ] Don't use [flashing GIFs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [ ] If the content were to be read as plain text, it still makes sense, and no information is missing.

### CODE REVIEW

File: docs/docusaurus.config.js [hunk 1]
```diff
@@ -72,9 +73,6 @@ const config = {
         docs: {
           path: "docs",
           routeBasePath: "docs",
-          admonitions: {
-            icons: "emoji",
-          },
```
Comment (kcpevey) [inline docs/docusaurus.config.js:77]: This throws an error with the new docusaurus version. Admittedly I dont know exactly what this does or how it works, so I could use some help here.

**Comment by kcpevey** [comment]
> Please let me know if you'd like me to refrain from pushing up to your branch in the future. 

I'll take all the help I can get - I always welcome collaboration :) 

Thank you so much for looking into this @pavithraes !

**Comment by kcpevey** [comment]
This is now ready for review. 

One thing that I do need resolved is the icon admonition (discussed in the description). If we're fine to remove, then its done. If it needs to stay, then I'll need help re-implementing it in a way that docusaurus v3 likes.

File: docs/docusaurus.config.js (hunk 1 above)
Comment (kcpevey) [reply docs/docusaurus.config.js:77]: This is the error:
```
[ERROR] Error: Unable to build website for locale en.
    at tryToBuildLocale (/Users/kcp/projects/nebari_project/nebari-docs2/docs/node_modules/@docusaurus/core/lib/commands/build.js:54:19)
    at async /Users/kcp/projects/nebari_project/nebari-docs2/docs/node_modules/@docusaurus/core/lib/commands/build.js:65:9
    at async mapAsyncSequential (/Users/kcp/projects/nebari_project/nebari-docs2/docs/node_modules/@docusaurus/utils/lib/jsUtils.js:21:24)
    at async Command.build (/Users/kcp/projects/nebari_project/nebari-docs2/docs/node_modules/@docusaurus/core/lib/commands/build.js:63:5) {
  [cause]: [Error [ValidationError]: "admonitions.icons" is not allowed] {
    _original: {
      path: 'docs',
      routeBasePath: 'docs',
      admonitions: { icons: 'emoji' },
      sidebarPath: '/Users/kcp/projects/nebari_project/nebari-docs2/docs/sidebars.js',
      sidebarCollapsible: true,
      editUrl: 'https://github.com/nebari-dev/nebari-docs/tree/main/docs',
      showLastUpdateAuthor: false,
      showLastUpdateTime: true
    },
    details: [
      {
        message: '"admonitions.icons" is not allowed',
        path: [ 'admonitions', 'icons' ],
        type: 'object.unknown',
        context: {
          child: 'icons',
          label: 'admonitions.icons',
          value: 'emoji',
          key: 'icons'
        }
      }
    ]
  }
}
```

Comment (kcpevey) [reply docs/docusaurus.config.js:77]: My understanding of this is that previously, we had to manually tell docusaurus to render emojis in markdown, and now this happens by default. After I remove these lines, the emojis still render so I believe this is fine to remove as no longer necessary.

**Comment by kcpevey** [comment]
I removed the icon admonition as I think its no longer necessary. This is ready for review.

**Comment by kcpevey** [comment]
I'll have to do conflict resolution again whenever we are close to merging.

**Comment by viniciusdc** [comment]
Hi @kcpevey as #502 what's the status on this PR, is it ready for review?

**Comment by kcpevey** [comment]
@viniciusdc Now that blocking PR has been merged, I'll resolve conflicts and get things working again. I'll ping you when its ready for final review.

**Comment by kcpevey** [comment]
@viniciusdc this is ready for a final look!

**Comment by viniciusdc** [comment]
@kcpevey I've noticed from the current built version of this PR, that some of the admonition colors and hyperlinks are not showing up properly, is this intended? 

### before:
image

### current PR build:
image

**Comment by kcpevey** [comment]
>I've noticed from the current built version of this PR, that some of the admonition colors and codeblocks styles are not showing up correctly, is this intended?

Hmmm. I'll have to look more closely. I did make a change to admonitions that may be having an effect here.

**Comment by viniciusdc** [comment]
Thats okay thank you for having a look into it @kcpevey -- winder if this might be due to docssaurus-sass plug-in

**Review by viniciusdc — APPROVED** [review]

**Pull request merged**